server/graphql/common/comment.ts

```diff
@@ -144,6 +144,13 @@ async function createComment(commentData, req): Promise<Comment> {
     },
   });
 
+  if (ExpenseId) {
+    const expense = await req.loaders.Expense.byId.load(ExpenseId);
+    if (remoteUser.isAdmin(expense.FromCollectiveId) && expense?.status === ExpenseStatus.INCOMPLETE) {
+      await expense.update({ status: ExpenseStatus.APPROVED });
+    }
+  }
+
```

Comment:
In `canMarkAsIncomplete` it seems a host admin can make an expense `INCOMPLETE` from `ERROR` and `PENDING`. In theory possible to have an expense transition like this:

`ERROR` | `PENDING`  -> `INCOMPLETE` by host admin -> `APPROVED` on submitter comment?

Maybe we should restrict the transition in `canMarkAsIncomplete` to `APPROVED` -> `INCOMPLETE` only?

Should we create an activity `COLLECTIVE_EXPENSE_APPROVED` here to mark the transition?

Reply:
Good point, you're totally correct about the PENDING to APPROVED here.
In this context, I think it makes sense to have ERROR -> INCOMPLETE -> APPROVED because will happen during processing.